Lib/ssl.py

```diff
@@ -628,8 +627,7 @@ def handshake_step(result):
             # http://stackoverflow.com/questions/24628271/exception-in-netty-io-netty-util-concurrent-blockingoperationexception
             # - we are doing this in the handler thread!
             return
-
-        self._sock._handle_channel_future(handshake, "SSL handshake")
+        self._sock._handle_channel_future(handshake, "SSL handshake", wait=True)
```

Comment:
I introduced the wait param here. The issue was that since Twisted is using non-blocking sockets (timeout of 0.0), the _handle_channel_future wouldn't wait for teh handshake to finish (https://github.com/riptano/jython/pull/5/files#diff-ff13dbb1a10b37c97a7a8d3daeec96a8R783) and the socket/protocol would continue its setup and wait forever since Jython would never throw a Handshake Error.